pkg/kubelet/kuberuntime/instrumented_services_test.go

```diff
@@ -61,6 +69,8 @@ func TestRecordOperation(t *testing.T) {
 	assert.HTTPBodyContains(t, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		mux.ServeHTTP(w, r)
 	}), "GET", prometheusURL, nil, runtimeOperationsDurationExpected)
+
+	registry.Reset()
```

Comment:
I don't think this is necessary since the test will terminate after evaluating this expression so we will not be using the registry anymore.

Reply:
It failed with `make test KUBE_RACE=-race KUBE_TIMEOUT=--timeout=600s GOFLAGS=-count=10 WHAT=./pkg/kubelet/kuberuntime/` before. The test is a little bit special, it runs with `-count=10`.

Reply:
Even if it is run 10 times, all of the tests should have independent registries which shouldn't collide with one another.

A potential reason why you were still seeing failures might be because you are using the `prometheus.DefaultRegisterer` in the handler which is shared between the tests. Although the library might be protecting against that, I haven't checked. But it might be worth checking again with my suggestion from above: https://github.com/kubernetes/kubernetes/pull/106289/files#r746633125

Reply:
Hi Damien, I think `registry.Reset()` is needed.

Even though registry is local,  metrics [RuntimeOperations](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/instrumented_services.go#L51-L52)  and [RuntimeOperationsDuration](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kuberuntime/instrumented_services.go#L51-L52) is global.

---
I tested that adding `metrics.RuntimeOperations.Reset()` would work if`registry.Reset()` was removed.
https://github.com/CatherineF-dev/kubernetes/commit/a8692c6e8fea86cddd63b9caab329acb2327ff4d

Reply:
Make sense, thank you for looking into that @CatherineF-dev :slightly_smiling_face: